net-im/telegram-desktop/telegram-desktop-4.10.3.ebuild

```diff
@@ -0,0 +1,239 @@
+	sys-libs/zlib:=[minizip]
+	virtual/opengl
+	!enchant? ( >=app-text/hunspell-1.7:= )
+	enchant? ( app-text/enchant:= )
+	jemalloc? ( dev-libs/jemalloc:=[-lazy-lock] )
+	!qt6? (
+		>=dev-qt/qtcore-5.15:5=
+		>=dev-qt/qtgui-5.15:5=[dbus,jpeg,png,wayland?,X?]
+		>=dev-qt/qtimageformats-5.15:5
+		>=dev-qt/qtnetwork-5.15:5[ssl]
+		>=dev-qt/qtsvg-5.15:5
+		>=dev-qt/qtwidgets-5.15:5[png,X?]
+		kde-frameworks/kcoreaddons:5
+		webkit? (
+			>=dev-qt/qtdeclarative-5.15:5
+			>=dev-qt/qtwayland-5.15:5
+		)
+	)
+	qt6? (
+		>=dev-qt/qtbase-6.5:6=[dbus,gui,network,opengl,wayland?,widgets,X?]
+		>=dev-qt/qtimageformats-6.5:6
+		>=dev-qt/qtsvg-6.5:6
+		wayland? ( >=dev-qt/qtwayland-6.5:6[compositor] )
+		webkit? (
+			>=dev-qt/qtdeclarative-6.5:6
+			>=dev-qt/qtwayland-6.5:6[compositor]
+		)
+		qt6-imageformats? (
+			>=dev-qt/qtimageformats-6.5:6=
+			${KIMAGEFORMATS_RDEPEND}
+		)
+	)
+	X? (
+		x11-libs/libxcb:=
+		x11-libs/xcb-util-keysyms
+	)
+"
+RDEPEND="${CDEPEND}
+	webkit? ( net-libs/webkit-gtk:4 )
+"
+DEPEND="${CDEPEND}
+	dev-cpp/cppgir
+	>=dev-cpp/ms-gsl-4
+	dev-cpp/range-v3
+"
+BDEPEND="
+	${PYTHON_DEPS}
+	dev-cpp/cppgir
+	>=dev-util/cmake-3.16
+	dev-util/gdbus-codegen
+	virtual/pkgconfig
+"
+# dev-libs/jemalloc:=[-lazy-lock] -> https://bugs.gentoo.org/803233
+
+PATCHES=(
+	"${FILESDIR}/tdesktop-4.2.4-jemalloc-only-telegram-r1.patch"
+	"${FILESDIR}/tdesktop-4.10.0-system-cppgir.patch"
+	"${FILESDIR}/tdesktop-4.10.3-fix-clang-libstdcxx.patch"
+)
+
+pkg_pretend() {
+	if has ccache ${FEATURES}; then
+		ewarn "ccache does not work with ${PN} out of the box"
+		ewarn "due to usage of precompiled headers"
+		ewarn "check bug https://bugs.gentoo.org/715114 for more info"
+		ewarn
+	fi
+}
+
+src_prepare() {
+	# Bundle kde-frameworks/kimageformats for qt6, since it's impossible to
+	#   build in gentoo right now.
+	if use qt6-imageformats; then
+		sed -e 's/DESKTOP_APP_USE_PACKAGED_LAZY/TRUE/' -i \
+			cmake/external/kimageformats/CMakeLists.txt || die
+		printf '%s\n' \
+			'Q_IMPORT_PLUGIN(QAVIFPlugin)' \
+			'Q_IMPORT_PLUGIN(HEIFPlugin)' \
+			'Q_IMPORT_PLUGIN(QJpegXLPlugin)' \
+			>> cmake/external/qt/qt_static_plugins/qt_static_plugins.cpp || die
+	fi
+
+	# kde-frameworks/kcoreaddons is bundled when using qt6, see:
+	#   cmake/external/kcoreaddons/CMakeLists.txt
+
+	# Happily fail if libraries aren't found...
+	find -type f \( -name 'CMakeLists.txt' -o -name '*.cmake' \) \
+		\! -path './cmake/external/expected/CMakeLists.txt' \
+		\! -path './cmake/external/qt/package.cmake' \
+		\! -path './Telegram/lib_webview/CMakeLists.txt' \
+		-print0 | xargs -0 sed -i \
+		-e '/pkg_check_modules(/s/[^ ]*)/REQUIRED &/' \
+		-e '/find_package(/s/)/ REQUIRED)/' || die
+	# Make sure to check the excluded files for new
+	# CMAKE_DISABLE_FIND_PACKAGE entries.
+
+	# Control QtDBus dependency from here, to avoid messing with QtGui.
+	if ! use dbus; then
+		sed -e '/find_package(Qt[^ ]* OPTIONAL_COMPONENTS/s/DBus *//' \
+			-i cmake/external/qt/package.cmake
```

Comment:
sed could fail if file not found, so missing "|| die" ?

Reply:
You're right. Thanks!